### PR TITLE
feat(dms): add new check `dms_endpoint_redis_tls_enabled`

### PR DESCRIPTION
**Pull request by danibarranqueroo**
### Context

This new check verifies whether AWS Database Migration Service (DMS) endpoints for Redis are configured with TLS (Transport Layer Security). The control fails if TLS is not enabled for the endpoint.

TLS provides end-to-end encryption for data in transit, ensuring secure communication between source and target databases during migration. Enabling TLS helps prevent unauthorized access to sensitive data, mitigating risks such as data breaches and interception.

RDS endpoints have a parameter inside the Redis Settings called SslSecurityProtocol. This allows to enable or disable the connection to a Redis target endpoint using Transport Layer Security (TLS). Valid values include plaintext and ssl-encryption. The default is ssl-encryption. The ssl-encryption option makes an encrypted connection. 

### Description

Added new check `dms_endpoint_redis_tls_enabled` with its unit tests.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [ ] Review if the code is being covered by tests.
- [ ] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/dms/dms_endpoint_redis_tls_enabled/dms_endpoint_redis_tls_enabled.metadata.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "Provider": "aws",
+  "CheckID": "dms_endpoint_redis_tls_enabled",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/dms/dms_endpoint_redis_tls_enabled/dms_endpoint_redis_tls_enabled.metadata.json]:
```suggestion
  "CheckID": "dms_endpoint_redis_in_transit_encryption_enabled",
```

File: prowler/providers/aws/services/dms/dms_service.py [hunk 2]
```diff
@@ -101,6 +104,7 @@ class Endpoint(BaseModel):
     region: str
     ssl_mode: str
     tags: Optional[list]
+    redis_tls_enabled: Optional[str]
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/dms/dms_service.py]:
```suggestion
    redis_ssl_protocol: Optional[str]
```

File: prowler/providers/aws/services/dms/dms_endpoint_redis_tls_enabled/dms_endpoint_redis_tls_enabled.metadata.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  "CheckTitle": "Check if DMS endpoints for Redis OSS have TLS enabled.",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/dms/dms_endpoint_redis_tls_enabled/dms_endpoint_redis_tls_enabled.metadata.json]:
```suggestion
  "CheckTitle": "Check if DMS endpoints for Redis OSS are encrypted in transit.",
```

File: prowler/providers/aws/services/dms/dms_endpoint_redis_tls_enabled/dms_endpoint_redis_tls_enabled.metadata.json (hunk 1 above)
Comment (danibarranqueroo) [reply prowler/providers/aws/services/dms/dms_endpoint_redis_tls_enabled/dms_endpoint_redis_tls_enabled.metadata.json]: Done!

File: prowler/providers/aws/services/dms/dms_service.py (hunk 2 above)
Comment (danibarranqueroo) [reply prowler/providers/aws/services/dms/dms_service.py]: Done!

File: prowler/providers/aws/services/dms/dms_endpoint_redis_tls_enabled/dms_endpoint_redis_tls_enabled.metadata.json (hunk 3 above)
Comment (danibarranqueroo) [reply prowler/providers/aws/services/dms/dms_endpoint_redis_tls_enabled/dms_endpoint_redis_tls_enabled.metadata.json]: Done!

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**